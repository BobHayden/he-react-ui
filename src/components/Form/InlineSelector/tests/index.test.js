import createTestContext from "react-cosmos-test/enzyme";
import fixture from "../fixtures/Default.fixture";

const { mount, getWrapper } = createTestContext({ fixture });

beforeEach(mount);

test("<InlineSelector /> rendered correctly with fixture", () => {
  const wrapper = getWrapper();
  Object.keys(fixture.props.options).forEach(key => {
    expect(wrapper.text()).toContain(fixture.props.options[key].label);
  });

  expect(wrapper.find("div")).toHaveLength(
    Object.keys(fixture.props.options).length + 2
  );

  expect(wrapper).toMatchSnapshot();
});
