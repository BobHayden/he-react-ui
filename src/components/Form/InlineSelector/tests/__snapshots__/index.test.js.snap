// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineSelector /> rendered correctly with fixture 1`] = `
<InlineSelector
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Any",
        "value": "1 ",
      },
      Object {
        "label": "AM",
        "value": "2",
      },
      Object {
        "label": "PM",
        "value": "3",
      },
      Object {
        "label": "After Hours",
        "value": "4",
      },
    ]
  }
  vertical={false}
>
  <div
    className=""
  >
    <div>
      <div
        className=""
        key="0"
        onClick={[Function]}
      >
        Any
      </div>
      <div
        className=""
        key="1"
        onClick={[Function]}
      >
        AM
      </div>
      <div
        className=""
        key="2"
        onClick={[Function]}
      >
        PM
      </div>
      <div
        className=""
        key="3"
        onClick={[Function]}
      >
        After Hours
      </div>
    </div>
  </div>
</InlineSelector>
`;
